pkg/yang/types_builtin.go

```diff
@@ -361,24 +300,32 @@ func (r YangRange) Contains(s YangRange) bool {
 // FromInt creates a Number from an int64.
 func FromInt(i int64) Number {
 	if i < 0 {
-		return Number{Kind: Negative, Value: uint64(-i)}
+		return Number{Negative: true, Value: uint64(-i)}
 	}
-	return Number{Kind: Positive, Value: uint64(i)}
+	return Number{Value: uint64(i)}
 }
 
 // FromUint creates a Number from a uint64.
 func FromUint(i uint64) Number {
-	return Number{Kind: Positive, Value: i}
+	return Number{Value: i}
 }
 
 // FromFloat creates a Number from a float64. Input values with absolute value
-// larger than MaxInt64/MinInt64 are converted into maxNumber/minNumber.
+// outside the boundaries specified for the decimal64 value specified in
+// RFC6020/RFC7950 are clamped down to the closest boundary value.
 func FromFloat(f float64) Number {
-	if f > float64(MaxInt64) {
-		return maxNumber
+	if f > MaxDecimal64 {
+		return Number{
+			Value:          FromInt(MaxInt64).Value,
+			FractionDigits: 1,
```

Comment:
i'm not sure I understand here. if `f > MaxDecimal64` shouldn't it be impossible to create this value - since we don't have any floating-point type in yang, and we're creating this `Number` in the context of a yang schema, what is the scenario in which it is not an error to create a greater-than-maximum-decimal64 `Number` that we're handling here?

Reply:
This function is only used by tests in goyang and ygot.

The primary reason why I chose to not return an error is because it makes it easier on tests that use this function, since you can just declare it within a struct declaration. Since ygot uses this, I can't make this function unexported without having to copy it into a different package, which I think is not ideal either.